lib/language-dataset-factory.js

```diff
@@ -37,9 +37,9 @@ export default class LanguageDatasetFactory {
    * @return {LanguageDataset} An instance of a language dataset.
    */
   static getDataset (languageID) {
-    let instance = this.instance
+    const instance = this.instance
     if (instance.sets.has(languageID)) {
-      let dataset = instance.sets.get(languageID)
+      const dataset = instance.sets.get(languageID)
```

Comment:
here the same question for using const - may be this rule uses `const` everywhere? then when `let` should be used then?
It is so strange for me :)

Reply:
`dataset` is an object, so `const dataset` guarantees that the `dataset` variable will always reference the same dataset object. But it does not prevent props and functions within the dataset object from being added, removed, or altered in any other way.

My understanding that for this rule ESLint does a syntactic analysis of the code, and if it sees a variable never being reassigned, it replaces `let` in this variable's definition to `const`. On my opinion, this should be a pretty trivial analysis and I also think thus it's pretty safe to apply, but I can be mistaken. What do you think?

Reply:
on the whole I am okay with this. I think regression testing will identify if it causes any problems and there has to be a full regression before the release.